twitter_stream_block.py

```diff
@@ -328,26 +331,43 @@ def _record_line(self, line):
                         report += ": {}".format(data['message'])
                     self._logger.debug(report)
 
-                    # Add a signal to the appropriate list
+                    # Calculate total limit for limit signals
                     if msg == "limit":
-                        with self._limit_lock:
-                            self._limit_signals.append(Signal(data))
-                    else:
-                        with self._diagnostic_lock:
-                            self._diagnostic_signals.append(Signal(data))
-                    
+                        # lock when calculating limit
+                        with self._get_result_lock('limit'):
+                            self._calculate_limit(data)
+
+                    # Add a signal to the appropriate list
+                    with self._get_result_lock(msg):
+                        self._result_signals[msg].append(Signal(data))
+
                     return
-                
+
+            # If we didn't return yet, the message is a regular tweet.
             self._logger.debug("It's a tweet!")
-            data = self.filter_results(json.loads(line.decode('utf-8')))
+            data = self.filter_results(data)
             if data:
-                tw = Signal(data)
-                with self._result_lock:
-                    self._result_signals.append(tw)
+                with self._get_result_lock('default'):
+                    self._result_signals['default'].append(Signal(data))
 
         except Exception as e:
             self._logger.error("Could not parse line: %s" % str(e))
 
+    def _get_result_lock(self, key):
+        with self._lock_lock:
+            return self._result_lock[key]
+
+    def _calculate_limit(self, data):
+        """ Calculate total limit count for limit signals """
+        track = data.get('limit', {}).get('track', 0)
+        if track > self._limit_count:
+            limit = track - self._limit_count
+            self._limit_count = track
+        else:
+            limit = 0
+        data['limit_count'] = limit
```

Comment:
Oh, never mind, I think it might already do that, just with the fields `track` and `limit_count` instead, that sound right? We'll just be sure to document if so